docs/asyncio_categorizing_functions.md

```diff
@@ -62,6 +62,27 @@ These are functions that are safe to run both in a thread and inside the event l
 
 There is no special annotation to mark functions as part of this category and care should be taken when using these functions from inside the event loop. When in doubt, look at their implementation.
 
+## Calling functions in async context
+
+There are mutliple ways to execute the different types of functions when being in the event loop (async context). All of the listed functions are methods of the HomeAssistant object.
+
+- Call sync functions: `async_add_executor_job`
+
+   They will be scheduled in the executor thread pool and can be awaited.
+- Call coroutines: `async_create_task`
+
+   Schedule coroutines. Returns the task object.
+- Call callback functions:
+
+   call directly without await.
+- Call async functions: `await ...`
+
+Additionally there are some legacy functions:
+
+- `add_job` - Same as `async_add_job` but without the possiblity to wait for it.
```

Comment:
This does not have the `async_` prefix. This is the only one that you are allowed to call from async context.